src/Button/Button.js

```diff
@@ -0,0 +1,102 @@
+import React, {Component, PropTypes} from 'react';
+import propTypes from './buttonPropTypes';
+import EnhancedButton from '../internal/EnhancedButton';
+import getStyles from './buttonStyles';
+import styleButtonChildren from './styleButtonChildren';
+
+import {
+  handleMouseDown,
+  handleMouseUp,
+  handleMouseLeave,
+  handleMouseEnter,
+  handleTouchStart,
+  handleTouchEnd,
+  handleKeyboardFocus,
+} from './buttonEventHandlers';
+
+class Button extends Component {
+  static muiName = 'Button';
+  static propTypes = propTypes;
+
+  static defaultProps = {
+    disabled: false,
+    fullWidth: false,
+    onKeyboardFocus: () => {},
+    onMouseDown: () => {},
+    onMouseEnter: () => {},
+    onMouseLeave: () => {},
+    onMouseUp: () => {},
+    onTouchEnd: () => {},
+    onTouchStart: () => {},
+    primary: false,
+    secondary: false,
+    type: 'raised',
+  };
+
+  static contextTypes = {
+    muiTheme: PropTypes.object.isRequired,
+  };
+
+  state = {
+    hovered: false,
+    keyboardFocused: false,
+    touched: false,
+    initialZDepth: 0,
+    zDepth: 0,
+  };
+
+  componentWillMount() {
+    const zDepth = (this.props.type === 'fab') ? 2 : 1;
```

Comment:
Oh, on the ternary? Personally prefer to keep it, as it makes it easier to read, and more consistent with an `if` statement, but I'll change it to maintain consistency of code style with other components.